system-test/storage.ts

```diff
@@ -2824,6 +2826,33 @@ describe('storage', () => {
         files![1].metadata.generation
       );
     });
+
+    it('should throw an error Precondition Failed on overwrite with version 0, then save file with and withour resumable', done => {
+      const fileName = `test-${Date.now()}.txt`;
+      bucketWithVersioning
```

Comment:
We can flatten this callback chain using Promise or async/await.

Reply:
@jkwlui  Converted to async await